Core/src/IfcGeometryConverter/SolidModelConverter.h

```diff
@@ -1326,7 +1327,7 @@ namespace OpenInfraPlatform
 					// sweeping curve is linear. assume any local z vector
 					local_z = carve::geom::VECTOR(0, 0, 1);
 					double dot_normal_local_z = dot((basisCurvePoints.at(1) - basisCurvePoints.at(0)), local_z);
-					if (abs(dot_normal_local_z) < 0.0001)
+					if (this->GeomSettings()->areEqual(abs(dot_normal_local_z), 0))

```

Comment:
```suggestion
					if (this->GeomSettings()->areEqual(abs(dot_normal_local_z), 0., 0.0001))
```

Always use doubles (0.) and stick with the original comparison delta.

Reply:
Resolved:  4bc3608